File: hashMap.js
class HashMap {
  constructor(hashTableSize) {
    this.hashTablesize = hashTableSize;
    this.hashArray = new Array(hashTableSize);
  }

  hashFunction = (key) => {
    let hash = 83;
    if (key == null) {
      return;
    }
    for (let char of key) {
      hash =
        (12223242526272829 * hash * char.charCodeAt(0)) % this.hashTablesize;
    }
    return hash;
  };

  get = (key) => {
    let index = this.hashFunction(key);
    return this.hashArray[index];
  };

  set = (key, value) => {
    let index = this.hashFunction(key);
    this.hashArray[index] = value;
  };

  contains = (key) => {
    let index = this.hashFunction(key);
    if (this.hashArray[index]) {
      return true;
    } else {
      return false;
    }
  };
}

File: hashMap.py
class HashMap:
    def __init__(self, hashTableSize):
        self.hashTableSize = hashTableSize
        self.hashTable = [None] * hashTableSize


    def hashFunction(self, key):
        hashValue = 83
        if key == None:
            return

        for char in key:
            hashValue = (12223242526272829 * hashValue * ord(char)) % self.hashTableSize

        return hashValue

    def setValue(self, key, value):
        index = self.hashFunction(key)
        self.hashTable[index] = value
    
    def getValue(self, key, value):
        index = self.hashFunction(key)
        return self.hashTable[index]

    def contains(self, key):
        index = self.hashFunction(key)
        if self.hashTable[index]:
            return True
        else:
            return False
